test/jit/test_pdt.py

```diff
@@ -224,3 +224,41 @@ def test_model_with_args(a, m):
         user_class = class_with_args(False)
         scripted_fn = torch.jit._script_pdt(test_model_with_args, example_inputs=[(10, user_class, ), (10.9, user_class, ), ])
         self.assertEqual(scripted_fn(100, class_with_args(True), ), test_model_with_args(100, class_with_args(True)))
+
+    def test_nn_module(self):
+        class Foo(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x):
+                assert x is not None
```

Comment:
All types can be compared against None. So this module compiles without PDT too. Please try some test cases that do not work with vanilla `script`.

Reply:
Added now.

Reply:
please remove this test case since it doesn't add additional test coverage for pdt.